### PR TITLE
add v510 benchmark doc

### PR DESCRIPTION
**Pull request by dbsid**
add v510 benchmark doc

### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

add v510 benchmark doc

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: benchmark/benchmark-sysbench-v5.1.0-vs-v5.0.2.md [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+| Threads   | v5.0.2 QPS   | v5.0.2 95% latency (ms)   | v5.1.0 QPS   | v5.1.0 95% latency (ms)   | QPS 提升   |
+|:----------|:----------|:----------|:----------|:----------|:----------|
+|150|15066.54|14.73|14829.31|14.73|-1.57%|
+|300|18535.92|24.83|17401.01|29.72|-6.12%|
+|600|22862.73|41.1|21923.78|44.98|-4.11%|
+|900|25286.74|57.87|24916.76|58.92|-1.46%|
+|1200|27566.18|70.55|27800.62|69.29|0.85%|
+|1500|28184.76|92.42|28679.72|86|1.76%|
+v5.1.0 对比 v5.0.2，Update Index 性能下降了 1.8%。
```
Comment (qiancai) [inline benchmark/benchmark-sysbench-v5.1.0-vs-v5.0.2.md:180]:
```suggestion

v5.1.0 对比 v5.0.2，Update Index 性能下降了 1.8%。
```

File: benchmark/benchmark-sysbench-v5.1.0-vs-v5.0.2.md [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+
+| Threads   | v5.0.2 QPS   | v5.0.2 95% latency (ms)   | v5.1.0 QPS   | v5.1.0 95% latency (ms)   | QPS 提升   |
+|:----------|:----------|:----------|:----------|:----------|:----------|
+|150|137732.27|1.86|158861.67|2|15.34%|
+|300|201420.58|2.91|238038.44|2.71|18.18%|
+|600|303631.52|3.49|428573.21|2.07|41.15%|
+|900|383628.13|3.55|464863.22|3.89|21.18%|
+|1200|391451.54|5.28|413656.74|13.46|5.67%|
+|1500|410276.93|7.43|471418.78|10.65|14.90%|
+v5.1.0 对比 v5.0.2，Point Select 性能提升了 19.4%。
```
Comment (qiancai) [inline benchmark/benchmark-sysbench-v5.1.0-vs-v5.0.2.md:151]:
```suggestion

v5.1.0 对比 v5.0.2，Point Select 性能提升了 19.4%。
```

**Review by qiancai — COMMENTED** [review]
@dbsid   PTAL at the comments. Thanks!

File: TOC.md [hunk 3]
```diff
@@ -11,8 +11,8 @@
   + [基本功能](/basic-features.md)
   + [实验特性](/experimental-features.md)
   + 性能测试报告
-    + [Sysbench 性能对比 - v5.0 对比 v4.0](/benchmark/benchmark-sysbench-v5-vs-v4.md)
-    + [TPC-C 性能对比 - v5.0 对比 v4.0](/benchmark/v5.0-performance-benchmarking-with-tpcc.md)
+    + [Sysbench 性能对比 - v5.1.0 对比 v5.0.2](/benchmark/benchmark-sysbench-v5.1.0-vs-v5.0.2.md)
+    + [TPC-C 性能对比 - v5.1.0 对比 v5.0.2](/benchmark/v5.1-performance-benchmarking-with-tpcc.md)
```
Comment (TomShawn) [inline TOC.md:15]: @dbsid @qiancai In test reports that compare different TiDB versions, the titles do not indicate the specific patch version, which is advised by @shuke987. 
```suggestion
    + [Sysbench 性能对比 - v5.1 对比 v5.0](/benchmark/benchmark-sysbench-v5.1-vs-v5.0.md)
    + [TPC-C 性能对比 - v5.1 对比 v5.0](/benchmark/v5.1-performance-benchmarking-with-tpcc.md)
```

File: benchmark/benchmark-sysbench-v5.1.0-vs-v5.0.2.md [hunk 4]
```diff
@@ -0,0 +1,194 @@
+---
+title: TiDB Sysbench 性能对比测试报告 - v5.1.0 对比 v5.0.2
+---
+
+# TiDB Sysbench 性能对比测试报告 - v5.1.0 对比 v5.0.2
+
+## 测试概况
+本次测试对比了 TiDB v5.1.0 和 v5.0.2 在 OLTP 场景下的 Sysbench 性能表现。结果显示，v5.1.0 相比于 v5.0.2，Point Select 场景性能提升了 19.4%，Read Write 和 Update Index 场景性能略有下降。
```
Comment (qiancai) [inline benchmark/benchmark-sysbench-v5.1.0-vs-v5.0.2.md:8]:
```suggestion
## 测试概况

本次测试对比了 TiDB v5.1.0 和 v5.0.2 在 OLTP 场景下的 Sysbench 性能表现。结果显示，v5.1.0 相比于 v5.0.2，Point Select 场景性能提升了 19.4%，Read Write 和 Update Index 场景性能略有下降。
```

File: benchmark/v5.1-performance-benchmarking-with-tpcc.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+| TiDB      | c5.4xlarge|     3     |
+| TPC-C  | c5.9xlarge|     1     |
+
+### 软件版本
+
+| 服务类型   | 软件版本    |
+|:----------|:-----------|
+| PD        | v5.0.2、v5.1   |
+| TiDB      | v5.0.2、v5.1   |
+| TiKV      | v5.0.2、v5.1   |
```
Comment (qiancai) [inline benchmark/v5.1-performance-benchmarking-with-tpcc.md:28]:
```suggestion
| PD        | v5.0.2、v5.1.0   |
| TiDB      | v5.0.2、v5.1.0   |
| TiKV      | v5.0.2、v5.1.0   |
```

**Review by qiancai — APPROVED** [review]

**Comment by shuke987** [comment]
LGTM

**Comment by shuke987** [comment]
LGTM

**Review by qiancai — APPROVED** [review]

**Review by TomShawn — APPROVED** [review]

File: benchmark/benchmark-sysbench-v5.1.0-vs-v5.0.2.md [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+#### TiDB 全局变量配置
+
+{{< copyable "sql" >}}
+
+```sql
+set global tidb_hashagg_final_concurrency=1;
+set global tidb_hashagg_partial_concurrency=1;
+```
+
+#### TiDB 全局变量配置
```
Comment (qiancai) [inline benchmark/benchmark-sysbench-v5.1.0-vs-v5.0.2.md:74]: There are two "#### TiDB 全局变量配置"

File: benchmark/v5.1-performance-benchmarking-with-tpcc.md [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+#### TiDB 全局变量配置
+
+{{< copyable "sql" >}}
+
+```sql
+set global tidb_hashagg_final_concurrency=1;
+set global tidb_hashagg_partial_concurrency=1;
+```
+
+#### TiDB 全局变量配置
```
Comment (qiancai) [inline benchmark/v5.1-performance-benchmarking-with-tpcc.md:76]: There are two "#### TiDB 全局变量配置" in this file.

File: benchmark/v5.1-performance-benchmarking-with-tpcc.md [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+## 测试方案
+
+1. 通过 TiUP 部署 TiDB v5.1.0 和 v5.0.2。
+
+2. 创建数据库tpcc: `create database tpcc;`
+2. 通过 tiup bench 导入 TPC-C 5000 Warehouse 数据: `tiup bench tpcc prepare --warehouse 5000 --db tpcc -H 127.0.0.1 -p 4000`。
+
+3. 运行 `tiup bench tpcc run -U root --db tpcc --host 127.0.0.1 --port 4000 --time 300s --warehouses 5000 --threads {{thread}}` 命令，通过 HAProxy 向 TiDB 加压，。
+
+4. 从结果中提取 New Order 的 tpmC 的数据。
```
Comment (qiancai) [inline benchmark/v5.1-performance-benchmarking-with-tpcc.md:98]:
```suggestion
1. 通过 TiUP 部署 TiDB v5.1.0 和 v5.0.2。

2. 创建数据库tpcc: `create database tpcc;`

3. 通过 tiup bench 导入 TPC-C 5000 Warehouse 数据: `tiup bench tpcc prepare --warehouse 5000 --db tpcc -H 127.0.0.1 -p 4000`。

4. 运行 `tiup bench tpcc run -U root --db tpcc --host 127.0.0.1 --port 4000 --time 300s --warehouses 5000 --threads {{thread}}` 命令，通过 HAProxy 向 TiDB 加压，。

5. 从结果中提取 New Order 的 tpmC 的数据。
```

File: benchmark/v5.1-performance-benchmarking-with-tpcc.md [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+
+1. 通过 TiUP 部署 TiDB v5.1.0 和 v5.0.2。
+
+2. 创建数据库tpcc: `create database tpcc;`
+
+3. 通过 tiup bench 导入 TPC-C 5000 Warehouse 数据: `tiup bench tpcc prepare --warehouse 5000 --db tpcc -H 127.0.0.1 -p 4000`。
+
+4. 运行 `tiup bench tpcc run -U root --db tpcc --host 127.0.0.1 --port 4000 --time 300s --warehouses 5000 --threads {{thread}}` 命令，通过 HAProxy 向 TiDB 加压。
+
+5. 从结果中提取 New Order 的 tpmC 的数据。
```
Comment (qiancai) [inline benchmark/v5.1-performance-benchmarking-with-tpcc.md:99]:
```suggestion
1. 通过 TiUP 部署 TiDB v5.1.0 和 v5.0.2。
2. 创建数据库tpcc: `create database tpcc;`
3. 通过 tiup bench 导入 TPC-C 5000 Warehouse 数据: `tiup bench tpcc prepare --warehouse 5000 --db tpcc -H 127.0.0.1 -p 4000`。
4. 运行 `tiup bench tpcc run -U root --db tpcc --host 127.0.0.1 --port 4000 --time 300s --warehouses 5000 --threads {{thread}}` 命令，通过 HAProxy 向 TiDB 加压。
5. 从结果中提取 New Order 的 tpmC 的数据。
```

File: benchmark/v5.1-performance-benchmarking-with-tpcc.md [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+raftstore.store-pool-size: 3
+readpool.storage.normal-concurrency: 10
+readpool.unified.max-thread-count: 20
+readpool.unified.min-thread-count: 5
+rocksdb.max-background-jobs: 8
+server.grpc-concurrency: 6
+storage.scheduler-worker-pool-size: 20
+server.enable-request-batch: false
+```
+
```
Comment (qiancai) [inline benchmark/v5.1-performance-benchmarking-with-tpcc.md:66]:
```suggestion
```

File: TOC.md [hunk 11]
```diff
@@ -11,8 +11,8 @@
   + [基本功能](/basic-features.md)
   + [实验特性](/experimental-features.md)
   + 性能测试报告
-    + [Sysbench 性能对比 - v5.0 对比 v4.0](/benchmark/benchmark-sysbench-v5-vs-v4.md)
-    + [TPC-C 性能对比 - v5.0 对比 v4.0](/benchmark/v5.0-performance-benchmarking-with-tpcc.md)
+    + [Sysbench 性能对比 - v5.1 对比 v5.0](/benchmark/benchmark/benchmark-sysbench-v5.1.0-vs-v5.0.2.md)
```
Comment (qiancai) [inline TOC.md:14]:
```suggestion
    + [Sysbench 性能对比 - v5.1 对比 v5.0](benchmark/benchmark-sysbench-v5.1.0-vs-v5.0.2.md)
```

File: TOC.md [hunk 12]
```diff
@@ -11,8 +11,8 @@
   + [基本功能](/basic-features.md)
   + [实验特性](/experimental-features.md)
   + 性能测试报告
-    + [Sysbench 性能对比 - v5.0 对比 v4.0](/benchmark/benchmark-sysbench-v5-vs-v4.md)
-    + [TPC-C 性能对比 - v5.0 对比 v4.0](/benchmark/v5.0-performance-benchmarking-with-tpcc.md)
+    + [Sysbench 性能对比 - v5.1 对比 v5.0](benchmark/benchmark-sysbench-v5.1.0-vs-v5.0.2.md)
```
Comment (qiancai) [inline TOC.md:14]:
```suggestion
    + [Sysbench 性能对比 - v5.1 对比 v5.0](/benchmark/benchmark-sysbench-v5.1.0-vs-v5.0.2.md)
```

**Pull request merged**